lib/edgehog_device_forwarder/termination_callbacks.ex

```diff
@@ -0,0 +1,36 @@
+# Copyright 2023 SECO Mind Srl
+# SPDX-License-Identifier: Apache-2.0
+
+defmodule EdgehogDeviceForwarder.TerminationCallbacks do
+  use GenServer
+
+  @spec add(pid, (-> any)) :: :ok
+  def add(pid, on_close) when is_function(on_close, 0) do
+    GenServer.cast(__MODULE__, {:add, pid, on_close})
+  end
+
+  @spec start_link(any) :: Supervisor.child()
+  def start_link(_value) do
+    GenServer.start_link(__MODULE__, %{}, name: __MODULE__)
+  end
+
+  @impl GenServer
+  def init(processes) do
+    {:ok, processes}
+  end
+
+  @impl GenServer
+  def handle_cast({:add, pid, on_close}, processes)
+      when is_function(on_close, 0) do
+    ref = Process.monitor(pid)
+    {:noreply, Map.put(processes, ref, on_close)}
```

Comment:
This GenServer sttae is not persisted, meaning that the ref <-> callback map will be lost if the `TerminationCallbacks` server goes down. Is this the intention, i.e. callback might not be executed for some processes?